MozillaSocial/mozilla-social-ios
# Analytics module

## Summary
* This PR adds an analytics module with a concrete implementation based on [Glean](https://github.com/mozilla/glean), and adds event to the `Discover` tab.

## Implementation Details
* Add `MoSoAnalytics` library. Its main components are:
    - `AnalyticsProvider`: provides analytics protocols and concrete implementations based on `Glean`
    - `BaseTracker`: a protocol for base tracking (`impression` and `engagement` events).
    - `GleanBaseTracker`: a concrete implementation of `BaseTracker` based on `Glean`
    - `DiscoverTracker`: a protocol with specialized methods to track events in the `Discover` tab
    - `MoSoDiscoverTracker`: a concrete implementation of `DiscoverTracker` based on any concrete implementation provided by `MoSoAnalytics` (currently, the one based on `Glean`)
* Update `DiscoverViewModel`, inject a `DiscoverTracker` instance and track engagement and impression events.

## Test Steps
* Build/run this branch, use the app (specifically the `Discover` tab) and make sure that events appear either in the console or in the online viewer. More specifically you should see events with the following identifiers:
    - `discover.recommendation.impression`: when a recommendation appears in the `Discover` list. You will likely see more than one, and they should have different `recommendationID` values
    - `discover.recommendation.open`: when tapping on a recommendation in the `Discover` list to view it in the web view
    - `discover.recommendation.share`: when you tap the share button in the `Discover` list
> **Note**
the event `discover.recommendation.bookmark` has been added to this PR as well, but as of now will not be fired because the `bookmark` action has not been implemented yet.

## PR Checklist:
- [ ] ~~Self Accessibility Review~~ Not applicable

## Screenshots
